onnxruntime/python/tools/transformers/models/stable_diffusion/optimize_pipeline.py

```diff
@@ -10,16 +10,14 @@
 # For example, the float32 ONNX pipeline is saved to ./sd-v1-5 directory, you can optimize and convert it to float16 like the following:
 #    python optimize_pipeline.py -i ./sd-v1-5 -o ./sd-v1-5-fp16 --float16
 #
-# Note that the optimized models are for CUDA Execution Provider. It might not run in other execution provider.
+# Note that the optimizations are carried out for CUDA Execution Provider at first, other EPs may not have the support for the fused opeartors.
```

Comment:
If that is the case, we shall add an option in optimize_pipeline.py like --execution_provider, so that fusion can be different for CUDA/Rocm.

Reply:
We shall not do it at the level of of `optimize_pipeline.py`, instead, do it in the fusion level. When call the fusion with `provider` hint, check the status of support. Otherwise, we will maintaining support matrices that will be scattered around all repo. This might be some other effort then.

Reply:
After some consideration, it seems that ROCm EP and handle the default optimization pretty well at the moment. The only needed options are `--float16 --use_multi_head_attention --use_raw_attention_mask`.